sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptableItem.java

```diff
@@ -0,0 +1,90 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ------------------------------------------------------------
+
+
+package com.azure.cosmos.encryption;
+
+import com.azure.cosmos.implementation.ItemDeserializer;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.encryption.EncryptionUtils;
+import com.azure.cosmos.implementation.guava25.base.Preconditions;
+
+/// <summary>
+/// Input type that can be used to allow for lazy decryption in the write path.
+/// </summary>
+/// <typeparam name="T">Type of item.</typeparam>
+/// <example>
+/// This example takes in a item, encrypts it and writes to Cosmos container.
+/// <code language="c#">
+/// <![CDATA[
+/// public class ToDoActivity{
+///     public string id {get; set;}
+///     public string status {get; set;}
+/// }
+///
+/// ToDoActivity test = new ToDoActivity()
+/// {
+///    id = Guid.NewGuid().ToString(),
+///    status = "InProgress"
+/// };
+///
+/// ItemResponse<EncryptableItem<ToDoActivity>> createResponse = await encryptionContainer
+// .CreateItemAsync<EncryptableItem<ToDoActivity>>(
+///     new EncryptableItem<ToDoActivity>(test),
+///     new PartitionKey(test.Status),
+///     EncryptionItemRequestOptions);
+///
+/// if (!createResponse.IsSuccessStatusCode)
+/// {
+///     //Handle and log exception
+///     return;
+/// }
+///
+/// (ToDoActivity toDo, DecryptionContext _) = await item.DecryptableItem.GetItemAsync<ToDoActivity>();
+/// ]]>
+/// </code>
+/// </example>
+
+// TODO moderakh add example
```

Comment:
NIT remove todo

Reply:
the public APIs haven't finalized yet. For that reason I would like to leave wiring the examples for later (to avoid re-do work), once the final APIs are finalized.